### PR TITLE
Users

### PR DESCRIPTION
**Pull request by MaksymZymyn**

### CODE REVIEW

**Review by sirius91191 — APPROVED** [review]

File: src/main/java/users/User.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.NoArgsConstructor;
+import utils.dao.HasId;
+import java.io.Serial;
+import java.io.Serializable;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class User implements Serializable, HasId {
+
```
Comment (yalukaiwo) [inline src/main/java/users/User.java:14]: I don't think it needts to be Serializable anymore

File: src/main/java/utils/environment/ConnectionDetails.java [hunk 2]
```diff
@@ -1,9 +1,9 @@
 package utils.environment;
 
-class ConnectionDetails {
+public class ConnectionDetails {
```
Comment (yalukaiwo) [inline src/main/java/utils/environment/ConnectionDetails.java:3]: this class doesn't have to be public. it is only used inside the package

**Review by yalukaiwo — CHANGES_REQUESTED** [review]
Users part is on me. The sql files are good. thought some changes will be implemented since i would like to do login by username rather than email. Redo all the Likes-related methods in LikesDAO. Given it is an independent database instance, requests to it should be sent from a separate DAO. No need for procfile yet

File: src/main/java/requests.http
Comment (yalukaiwo) [inline src/main/java/requests.http:1]: unnecessary file

File: src/main/java/servlets/ParameterServlet.java
Comment (yalukaiwo) [inline src/main/java/servlets/ParameterServlet.java:1]: Also an unnecesary file. unused servlet which will not be used.

File: src/main/java/servlets/UsersRedirectServlet.java
Comment (yalukaiwo) [inline src/main/java/servlets/UsersRedirectServlet.java:1]: Unused servlet.

File: src/main/resources/db/migration/V0001_create_tables.sql [hunk 3]
```diff
@@ -0,0 +1,37 @@
+CREATE TABLE users (
+     id UUID PRIMARY KEY,
+     username TEXT NOT NULL UNIQUE,
+     full_name TEXT NOT NULL,
+     password VARCHAR(255) NOT NULL,
+     picture TEXT,
+     created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+     updated_at TIMESTAMP
+);
```
Comment (yalukaiwo) [inline src/main/resources/db/migration/V0001_create_tables.sql:9]: table users doesn't have created_at and updated_at

File: src/main/resources/db/migration/V0001_create_tables.sql [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+     id SERIAL PRIMARY KEY,
+     user_from UUID NOT NULL,
+     user_to UUID NOT NULL,
+     value BOOLEAN NOT NULL,
+     created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+     CONSTRAINT user_likes_users_from_fk FOREIGN KEY (user_from) REFERENCES users (id) ON DELETE CASCADE,
+     CONSTRAINT user_likes_users_to_fk FOREIGN KEY (user_to) REFERENCES users (id) ON DELETE CASCADE
+);
+
+CREATE TABLE chats (
```
Comment (yalukaiwo) [inline src/main/resources/db/migration/V0001_create_tables.sql:21]: this table is unnecessary

File: src/main/resources/templates/auth.ftl
Comment (yalukaiwo) [inline src/main/resources/templates/auth.ftl:1]: there is login.ftl already which serves the same purpose

File: src/main/java/likes/LikesDAO.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+import java.sql.SQLException;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+public class LikesDAO implements DAO<Like>{
+
+    private Like createLikeFromResultSet(ResultSet resultSet) throws SQLException {
```
Comment (yalukaiwo) [inline src/main/java/likes/LikesDAO.java:17]: fromResultSet in my opinion should be a static function in Like.java. Look at User.java fromRS function for reference

File: src/main/java/likes/Like.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import java.io.Serial;
+import java.io.Serializable;
+
+public class Like implements Serializable, HasId {
+
+    @Serial
+    private static final long serialVersionUID = 1L;
+    private final int id;
+    private final int user_from;
+    private final int user_to;
```
Comment (yalukaiwo) [inline src/main/java/likes/Like.java:14]: user_from and user_to are UUID, not int

File: src/main/java/database/DatabaseSetup.java
Comment (yalukaiwo) [inline src/main/java/database/DatabaseSetup.java:1]: unnesessary file for now

**Review by yalukaiwo — CHANGES_REQUESTED** [review]
Sorry for torturing you with changes. new comments added

**Review by sirius91191 — APPROVED** [review]

**Review by sirius91191 — APPROVED** [review]

**Review by sirius91191 — APPROVED** [review]

**Review by sirius91191 — APPROVED** [review]

File: src/main/java/likes/LikeDAO.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+        return filteredLikes;
+    }
+
+    @Override
+    @SneakyThrows(SQLException.class)
+    public void delete(UUID id) {
+
+        try (Connection connection = Database.connect()) {
+             String sql = "DELETE FROM likes WHERE userId = ?";
```
Comment (yalukaiwo) [inline src/main/java/likes/LikeDAO.java:98]: user_from

File: src/main/java/likes/LikeService.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+    public List<Like> getAll() {
+        return likeDAO.getAll();
+    }
+
+    public Optional<Like> get(UUID id) {
+        return likeDAO.get(id);
+    }
+
+    public boolean hasBeenLiked(UUID id, String user_to) {
```
Comment (yalukaiwo) [inline src/main/java/likes/LikeService.java:28]: UUID user_to

File: src/main/resources/db/migration/V002__initial_data.sql [hunk 9]
```diff
@@ -0,0 +1,14 @@
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
```
Comment (yalukaiwo) [inline src/main/resources/db/migration/V002__initial_data.sql:1]: remove uuid-ossp

File: src/main/resources/db/migration/V002__initial_data.sql [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+-- Inserting data into the 'users' table
+INSERT INTO users (id, username, full_name, picture, password) VALUES
```
Comment (yalukaiwo) [inline src/main/resources/db/migration/V002__initial_data.sql:4]: remove manual id set

File: src/main/resources/templates/auth.ftl
Comment (yalukaiwo) [inline src/main/resources/templates/auth.ftl:1]: remove file

**Review by yalukaiwo — CHANGES_REQUESTED** [review]

File: src/main/java/servlets/LikeServlet.java [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+                            } catch (IOException e) {
+                                throw new RuntimeException(e);
+                            }
+                        },
+                        () -> Auth.renderUnregistered(resp)
+                );
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
```
Comment (yalukaiwo) [inline src/main/java/servlets/LikeServlet.java:75]: no need to redirect to get on post. either don't create an endpoint or throw 400 error

**Review by yalukaiwo — APPROVED** [review]
approved. some minor fixes to be done

**Pull request merged**